### PR TITLE
Made ruby plugin support empty package names

### LINKED ISSUE #4984: Ruby does not support empty proto package name
From http://stackoverflow.com/questions/35114226/troubles-in-using-grpc-python-server-with-ruby-client

The error looks like it is assuming there is a package name (because of the leading period):
```
in `check_status': 12:Method "Serve" of service ".A" not implemented! (GRPC::BadStatus)
```

Related: #4824

### PR DESCRIPTION
**Pull request by murgatroid99**
This fixes #4984.

I'm pretty sure this is the only case where having an empty package actually could cause a problem. Other places use a function that explicitly removes leading periods.

### CODE REVIEW

File: src/compiler/ruby_generator.cc [hunk 1]
```diff
@@ -98,7 +98,7 @@ void PrintService(const ServiceDescriptor *service, const grpc::string &package,
   out->Print("self.marshal_class_method = :encode\n");
   out->Print("self.unmarshal_class_method = :decode\n");
   std::map<grpc::string, grpc::string> pkg_vars =
-      ListToDict({"service.name", service->name(), "pkg.name", package, });
+      ListToDict({"service_full_namename", service->full_name()});
```
Comment (jtattermusch) [inline src/compiler/ruby_generator.cc]: "namename"? Where is this used?

Comment (murgatroid99) [inline src/compiler/ruby_generator.cc]: That was a typo. And I did forget to change the following line.

**Comment by jtattermusch** [comment]
LGTM.

**Comment by jtattermusch** [comment]
@murgatroid99  does this need to be in the release?

**Comment by murgatroid99** [comment]
It's not directly related to the packaging stuff, so I guess not.

**Pull request merged**